### PR TITLE
Fixes for 6.5 kernel

### PR DESCRIPTION
**Pull request by ShadowCurse**
## Changes
Couple fixes for the CI for both aarch64 and x86. 

## License Acceptance

By submitting this pull request, I confirm that my contribution is made under
the terms of the Apache 2.0 license. For more information on following Developer
Certificate of Origin and signing off your commits, please check
[`CONTRIBUTING.md`][3].

## PR Checklist

- [ ] If a specific issue led to this PR, this PR closes the issue.
- [ ] The description of changes is clear and encompassing.
- [ ] Any required documentation changes (code and docs) are included in this
  PR.
- [ ] API changes follow the [Runbook for Firecracker API changes][2].
- [ ] User-facing changes are mentioned in `CHANGELOG.md`.
- [ ] All added/changed functionality is tested.
- [ ] New `TODO`s link to an issue.
- [ ] Commits meet
  [contribution quality standards](https://github.com/firecracker-microvm/firecracker/blob/main/CONTRIBUTING.md#contribution-quality-standards).

______________________________________________________________________

- [ ] This functionality cannot be added in [`rust-vmm`][1].

[1]: https://github.com/rust-vmm
[2]: https://github.com/firecracker-microvm/firecracker/blob/main/docs/api-change-runbook.md
[3]: https://github.com/firecracker-microvm/firecracker/blob/main/CONTRIBUTING.md

### CODE REVIEW

File: tests/integration_tests/functional/test_cpu_template_helper.py
Comment (zulinx86) [inline tests/integration_tests/functional/test_cpu_template_helper.py:453]: How about excluding it from CPU config?
https://github.com/firecracker-microvm/firecracker/blob/5d3a13d225b452f36e23f5a616cc190f3b30a5cf/src/vmm/src/vstate/vcpu/aarch64.rs#L239

Comment (ShadowCurse) [reply tests/integration_tests/functional/test_cpu_template_helper.py:453]: I really don't want FC to have any filters for cpu registers. If we will, I think It will become an issue by itself over time. Same as cpu normalization for x86.

Comment (zulinx86) [reply tests/integration_tests/functional/test_cpu_template_helper.py:453]: We already have it both on x86 and aarch64. Otherwise, fingerprint verification will keep false alarming and users need to filter them as you do in the test. I guess moving the filter from Firecracker to CPU template helper tool is outside of this PR. So let's filter it in Firecracker as of now.

Comment (ShadowCurse) [reply tests/integration_tests/functional/test_cpu_template_helper.py:453]: This is exactly what I don't want for aarch64 is to repeat filtering as it is done in x86. Considering that this register does not cause any problems by itself (snapshot tests pass with it not filtered out), I think it is our tests should have a knowledge what to test. Also fingerprint verification will be alarming in both cases when new register suddenly will become available. I can add it to the `cpu-template-helper-tool` if you want.

Comment (zulinx86) [reply tests/integration_tests/functional/test_cpu_template_helper.py:453]: > Considering that this register does not cause any problems by itself (snapshot tests pass with it not filtered out), I think it is our tests which should have a knowledge of what to test and what not to test.

I don't think so. I believe the CPU template helper should filter them out for users.

> I can add it to the cpu-template-helper-tool if you want.

I'll take the task because I'm working on another refactoring stuff.
OK. Please leave this PR as it is. I'll remove the filters in test in the refactor.

**Review by zulinx86 — DISMISSED** [review]

**Review by zulinx86 — COMMENTED** [review]
https://github.com/firecracker-microvm/firecracker/blob/main/docs/cpu_templates/cpu-template-helper.md#appendix

Could you please add them to the ignore list of the doc.

File: tests/integration_tests/functional/test_cpu_template_helper.py
Comment (ShadowCurse) [reply tests/integration_tests/functional/test_cpu_template_helper.py:453]: > I don't think so. I believe the CPU template helper should filter them out for users. As maintainer, we should decide which registers should be cared by users in the fingerprint verification process. Otherwise, the helper tool is not helpful despite its name.

I am totally fine with template helper filtering out things. Just not FC itself.

**Comment by ShadowCurse** [comment]
Added `HV_X64_MSR_TSC_INVARIANT_CONTROL` to the doc. Did not include aarch64 register, because it is filtered out in tests and not in FC.

File: tests/integration_tests/functional/test_cpu_template_helper.py
Comment (zulinx86) [reply tests/integration_tests/functional/test_cpu_template_helper.py:453]: yep, I'm planning to move them all from FC to the helper tool.

**Review by zulinx86 — APPROVED** [review]

**Review by sudanl0 — APPROVED** [review]

File: tests/integration_tests/functional/test_cpu_template_helper.py
Comment (zulinx86) [reply tests/integration_tests/functional/test_cpu_template_helper.py:453]: While doing the refactoring, I found that KVM_GET_MSRS call fails for some of MSR index list returned by KVM_GET_MSR_INDEX_LIST call (e.g., MSR_ARCH_PERFMON_FIXED_CTR0). That means in any cases, we need to have such a list on Firecracker side.

Comment (ShadowCurse) [reply tests/integration_tests/functional/test_cpu_template_helper.py:453]: This is an x86 part of code right? For aarch64 you should be able to move this one filtered out reg into the helper tool.

Comment (zulinx86) [reply tests/integration_tests/functional/test_cpu_template_helper.py:453]: Why do we want to handle it differently between x86 and aarch64?

**Pull request merged**
